lua/Events.lua

```diff
@@ -1,10 +1,14 @@
+local triggerables = {}
+
 -- Adds event handling to an object's instance
 function library.addEventHandlers(obj)
   -- Does nothing when obj is null or already has events
   if (not obj) or (obj.onEvent and obj.triggerEvent) then
     return false
   end
 
+  triggerables[#triggerables+1] = obj

```

Comment:
For this, I would be careful and keep track of the index on a separate variable, then loop indexes `1..nTriggerables`, this would 100% guarantee no entry in that table ever gets overriden by another entry in case something gets deleted.

Reply:
We can change that to a separate variable, but it will not increase much the performance as these functions were not intended to be call in loop/update.

About the loop, agree. I don't know why I went for `pairs` instead of `i=1,#triggerables` format.